blocks/subscriptions-block/components/usePaywall.js

```diff
@@ -0,0 +1,121 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useFusionContext } from 'fusion:context';
+import getProperties from 'fusion:properties';
+
+import Identity from '@arc-publishing/sdk-identity';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+
+const usePaywall = () => {
+  const { arcSite, globalContent } = useFusionContext();
+  const { api } = getProperties(arcSite);
+
+  const [isPaywalled, setIsPaywalled] = useState(false);
+  const [campaignCode, setCampaignCode] = useState();
+  const [results, setResults] = useState();
+  const [triggeredRule, setTriggeredRule] = useState();
+  const [otherTriggeredRules, setOtherTriggeredRules] = useState([]);
+  const [isSignwall, setIsSignwall] = useState();
+
+  // eslint-disable-next-line no-underscore-dangle
+  const rules = window?.ArcP?._rules;
+
+  const apiOrigin = api?.retail?.origin;
+
+  useEffect(() => {
+    const runPaywall = async () => {
+      setResults(
+        await window?.ArcP?.run({
+          apiOrigin,
```

Comment:
I created a test block to be able to run this hook and the `apiOrigin` was causing some errors as the SDK doesn't expect the origin URL to have `https://` at the start

This is the line in the Subs code ->
```t = "".concat(r.apiOrigin ? "https://" + r.apiOrigin : "", "/sales/public/v1/entitlements")```

Reply:
Adding a replace to strip this out.  I can't remove it from the origin url because that's being used by another hook.